torch/nested/__init__.py

```diff
@@ -302,3 +304,87 @@ def narrow(tensor: Tensor, dim: int, start: Union[int, Tensor], length: Union[in
         raise RuntimeError(f"Specified layout is unsupported for nested narrow: {layout}")
 
     return nt
+
+
+def nested_tensor_from_jagged(
```

Comment:
Will there be methods for me to pass in the  min_seqlen, max_seqlen manually? 
The goal is to avoid calculating it from the offsets(gpu) and syncing it to cpu.

Reply:
With the new infrastructure, it would be possible have a separate API to associate lengths/offsets with cached quantities. 
Tentatively it would look something like `get_global_nt_state().uf.get_metadata(offsets)["max_seqlen"]  = max_seqlen`

Reply:
We could probably wrap this something simpler: `nt_set_metadata(offsets, "max_seqlen", max_seqlen)` (name bikesheddable)

Reply:
@soulitzer so that means we'd just calculate this once and then cache it automatically? So then @YuqingJ doesn't have to store it, because we calculate it as needed.

Reply:
I think that is an option, but if I understand correctly @YuqingJ wants to compute it and store it into the cache ahead of time instead of incurring a sync in the middle of the program (though I'm not sure how much of a concern this given that this cost would be amortized over all places we need to use it later).

Reply:
> though I'm not sure how much of a concern this given that this cost would be amortized over all places we need to use it later

wanted to point out that depending on the scenario, this might not necessarily be true; e.g. with PT2 you sometimes have a lot of CPU latency at the end of a compiled section (e.g. doing AccumulateGrad, or creating views) - you could imagine that if the sync is triggered near the end of the compiled section, it might cause GPU idleness at the end of the compiled section when the views run

Reply:
based on the above discussion, I'm going to punt on this until @soulitzer's nested int work lands

Reply:
based on the above discussion, I'm going to punt on this until @soulitzer's nested int work lands